ubuntu_18.04.sh

```diff
@@ -1,24 +1,166 @@
 #!/bin/bash
-# This script builds the repo version of PyNE (with the MOAB optional
-# dependency) from scratch on Ubuntu 18.04. The folder $HOME/opt is created
-# and PyNE is installed within.
-#
-# Run this script from any directory by issuing the command where <version>
-# is either "dev" or "stable":
-# $ ./ubuntu_18.04.sh <version>
-# After the build finishes run:
-#  $ source ~/.bashrc
-# or open a new terminal.
-
 # Use package manager for as many packages as possible
-apt_package_list="software-properties-common wget \
-             build-essential git cmake vim emacs gfortran libblas-dev \
-             python-pip liblapack-dev libeigen3-dev libhdf5-dev autoconf libtool"
+apt_package_list="software-properties-common \
+                  wget \
+                  build-essential \
+                  git \
+                  cmake \
+                  gfortran \
+                  libblas-dev \
+                  liblapack-dev \
+                  libeigen3-dev \
+                  libhdf5-dev \
+                  autoconf \
+                  libtool \
+                  hdf5-tools"
 
-pip_package_list="numpy scipy cython nose tables matplotlib jinja2 \
-                  setuptools future"
+pip_package_list="numpy \
+                  scipy \
+                  cython \
+                  nose \
+                  tables \
+                  matplotlib \
+                  jinja2 \
+                  setuptools \
+                  future"
 
+# hdf5 std directory
 hdf5_libdir=/usr/lib/x86_64-linux-gnu/hdf5/serial
 
+#!/bin/bash
+# This script contains common code for building PyNE on various Debian-derived systems
+#
+source utils.sh
+
+# system update
+sudo apt-get -y update
+sudo apt-get install -y ${apt_package_list}
+python -m pip install --user --upgrade pip
+pip install --user ${pip_package_list}
+
+install_dir=${HOME}/opt
+mkdir -p ${install_dir}
+
+# need to put libhdf5.so on LD_LIBRARY_PATH
+echo "export LD_LIBRARY_PATH=${hdf5_libdir}" >> ~/.bashrc
+echo "export LIBRARY_PATH=${hdf5_libdir}" >> ~/.bashrc
+
+source ~/.bashrc
+
+############
+### MOAB ###
+############
+
+# pre-setup
+cd ${install_dir}
+check_repo moab
+mkdir -p moab
+cd moab
+
+# clone and version
+git clone --branch Version5.1.0 --single-branch https://bitbucket.org/fathomteam/moab moab-repo
+cd moab-repo
+mkdir -p build
+cd build
+
+# cmake, build and install
+cmake ../ -DENABLE_HDF5=ON -DHDF5_ROOT=${hdf5_libdir} \
+            -DBUILD_SHARED_LIBS=ON \
+            -DENABLE_PYMOAB=ON \
+            -DENABLE_BLASLAPACK=OFF \
+            -DENABLE_FORTRAN=OFF \
+            -DCMAKE_INSTALL_PREFIX=${install_dir}/moab
+make
+make install
+
```

Comment:
delete the moab and build directories after building?

Reply:
not sure about that one.

We definitely could, we did't before...